src/ansys/fluent/core/services/field_data.py

```diff
@@ -838,6 +838,134 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class _FieldDataBase:
+    def __init__(self, i_d, data):
+        self._data = data
+        self._id = i_d
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def surface_id(self):
+        return self._id
+
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(_FieldDataBase):
+    class _ScalarData:
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        self._data = []
+        for _data in data:
+            self._data.append(ScalarFieldData._ScalarData(_data))
+        super().__init__(i_d, self._data)
```

Comment:
@prmukherj we can remove all lives except for the `super` line and we can pass a list comprehension as the second (data) argument.

Reply:
Thank you @seanpearsonuk. This is addressed.